src/parser/SparqlParser.h

```diff
@@ -15,14 +15,12 @@
 
 using std::string;
 
-enum QueryType { CONSTRUCT_QUERY, SELECT_QUERY };
-
 // A simple parser of SPARQL.
 // No supposed to feature the complete query language.
 class SparqlParser {
  public:
   explicit SparqlParser(const string& query);
-  ParsedQuery parse();
+  static ParsedQuery parseQuery(std::string_view query);
```

Comment:
The Docstring above this class is now out of sync.

rather "The SPARQL parser used by QLever. The actual parsing is delegated to a parser that is based on ANTLR4, which supports the complete SPARQL 1.1 grammar. The only thing that is still parsed manually by this class are FILTER clauses, because QLever doesn't support arbitrary expressions in filters yet".

Reply:
I disagree with writing `supports the complete SPARQL 1.1 grammar`. This may be mistaken. At lexing/parsing level this statement is true. But there are still some features that are not implemented/supported in the Visitor.